runtime/jcl/common/thread.cpp

```diff
@@ -569,18 +547,116 @@ Java_java_lang_Thread_registerNatives(JNIEnv *env, jclass clazz)
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	jobjectArray result = (jobjectArray)VM_VMHelpers::createLocalRef(env, currentThread->scopedValueCache);
+	VM_VMAccess::inlineExitVMToJNI(currentThread);
+
+	return result;
+}
+
+/**
+ * static native void setScopedValueCache(Object[] cache);
+ *
+ * Set the scoped value cache.
+ *
+ * @param env instance of JNIEnv.
+ * @param unusedClass
+ * @param cache the scoped value cache object array.
+ * @return void
+ */
+void JNICALL
+Java_java_lang_Thread_setScopedValueCache(JNIEnv *env, jclass unusedClass, jobjectArray cache)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	currentThread->scopedValueCache = J9_JNI_UNWRAP_REFERENCE(cache);
+	VM_VMAccess::inlineExitVMToJNI(currentThread);
+}
+#else /* JAVA_SPEC_VERSION >= 20 */
+/**
+ * static native Object[] extentLocalCache();
+ *
+ * Get the extent local cache.
+ *
+ * @param env instance of JNIEnv.
+ * @param unusedClass
+ * @return jobjectArray
+ */
+jobjectArray JNICALL
+Java_java_lang_Thread_extentLocalCache(JNIEnv *env, jclass unusedClass)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	jobjectArray result = (jobjectArray)VM_VMHelpers::createLocalRef(env, currentThread->extentLocalCache);
+	VM_VMAccess::inlineExitVMToJNI(currentThread);
+
+	return result;
+}
+
+/**
+ * static native void setExtentLocalCache(Object[] cache);
+ *
+ * Set the extent local cache.
+ *
+ * @param env instance of JNIEnv.
+ * @param unusedClass
+ * @param cache the extent local cache object array.
+ * @return void
+ */
+void JNICALL
+Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobjectArray cache)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	VM_VMAccess::inlineEnterVMFromJNI(currentThread);
+	currentThread->extentLocalCache = J9_JNI_UNWRAP_REFERENCE(cache);
+	VM_VMAccess::inlineExitVMToJNI(currentThread);
+}
+#endif /* JAVA_SPEC_VERSION >= 20 */
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+/**
+ * static native Object findScopedValueBindings();
+ *
+ * Find the most recent scoped value bindings in the stack.
+ * TODO: Complete the function description.
+ *
+ * @param env instance of JNIEnv.
+ * @param unusedClass
+ * @return jobject
+ */
+jobject JNICALL
+Java_java_lang_Thread_findScopedValueBindings(JNIEnv *env, jclass unusedClass)
+{
+	/* TODO: Implement. See https://github.com/eclipse-openj9/openj9/pull/16678. */
+	return NULL;
+}
 
-/* static native void ensureMaterializedForStackWalk(Object o);
+/**
+ * static native void ensureMaterializedForStackWalk(Object o);
  *
  * This is expected to invoke JVM_EnsureMaterializedForStackWalk and ensure
  * that the stackwalk code sees a materialized value.
  *
- * @param env instance of JNIEnv
+ * @param env instance of JNIEnv.
  * @param unusedClass
- * @param obj the object to be checked
+ * @param obj the object to be checked.
+ * @return void
```

Comment:
Please don't make these changes. Periods are not wanted for `@param` entries and `@return` isn't appropriate for a function that doesn't return anything.